### ISSUE TITLE
array plot throws hard to understand error when array contains Nil

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#3725]
ArrayedCollection plot used to throw hard to understand errors when it contained a nil value. It should warn and plots anyway, replacing the value by zero.

```supercollider
[1, 2, 3, nil, nil].plot // ERROR: Message '<' not understood.
[nil, 1, 2, 3].plot // ERROR: Message '-' not understood.
```

**Comment by brianlheim** [comment]
From experience with other plotting utilities, I expect at least two things from this kind of API: being able to graph with missing values, and being able to specify the value that replaces nil. There should be no default replacement value IMO, as different domains and cases will have different ideas of default (0 vs 1 is the obvious tradeoff).

I don't know how difficult it is to add support for graphing with nil values; the minimum work for now as I see it would be to add the replacement value parameter and then produce a clearer [error] message for nil.

**Comment by seansay** [comment]
Yes, we should fix this...right now the error message is hard to follow.

I agree with @brianlheim fallback to zero isn't always the use case for the problem its dealing with, so maybe make the replacement value an argument for `plot()` make more sense.

**Comment by telephon** [comment]
can you be a little more specific? What is a clearer error message? I am a little hesitant to add another argument to the `plot` method.

**Comment by telephon** [comment]
I'd actually prefer to just throw an error and not plot at all.

**Comment by brianlheim** [comment]
A clearer error message would be something like "cannot plot: non-numeric value at index %".

**Comment by brianlheim** [comment]
Plotting is a complex request and it's not uncommon for such a function to have tons of arguments. see, for example, the plotting methods in R and mathplotlib in Python. Either behavior is fine with me, so long as it's something we could extend later.
